components/module-components/SearchAndFilter/Filter.tsx

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-unused-vars */
+
+import React, { useState, useEffect } from 'react';
+import style from './filter.module.scss';
+
+export enum UserRole {
+  Admin = 'Admin',
+  Member = 'Member',
+  ContentManager = 'Content Manager',
+}
+
+interface FilterProps {
+  handleRoleChange: (roles: UserRole[]) => void;
+}
+
+const Filter = ({ handleRoleChange }: FilterProps) => {
+  const [selectedOption, setSelectedOption] = useState<UserRole | ''>('');
+
+  useEffect(() => {
+    if (selectedOption === '') {
+      handleRoleChange(Object.values(UserRole));
+    } else {
+      handleRoleChange([selectedOption]);
+    }
+  }, [selectedOption, handleRoleChange]);
```

Comment:
we can memoize the UserRole options since most likely they arent going to change.
and use a ternary operator here instead the if else